### DISCUSSION TITLE
AccelStepper

### DISCUSSION THREAD

**Discussion by doceave** [micropython/micropython#9978] [Libraries & Drivers]
I am hoping to drive a bipolar stepper motor using a TB6600 driver. Doing this using the Arduino AccelStepper library was straightforward. I am having difficulty implementing the Micropython port of the same by pedromneto97: https://github.com/pedromneto97/AccelStepper-MicroPython

Would someone kindly post an example of driver initiation and some basic motor movements? The ability to accelerate to a speed and then slow to a stop at a specific location would illustrate things well.

Many thanks in advance.

**Comment by doceave** [comment]
I have figured it out... for this driver the instructions would take the form of:

```
print('reset to position')
stepper.set_current_position(0)
stepper.set_acceleration(0.5)
stepper.set_max_speed(10)
stepper.move_to(100)
while stepper.run():
    print(stepper.current_position())
    stepper.run()
```

**Reply by doceave** [reply]
What I was really seeking, however, is a non-blocking implementation such as the one implemented here by Curious Scientist: https://youtu.be/CC4nBDcz6Xs

This without having to invoke uasyncio :/

**Comment by cmregalado** [comment]
@doceave were you able to get anywhere with this? I am trying to do the same but am having trouble getting my project off the ground.
